File: C++/linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        
        if (head != NULL and head->next != NULL) {
            ListNode* fptr = head;
            ListNode* sptr = head;

            while (fptr != NULL and fptr->next != NULL) {
                fptr = fptr->next->next;
                sptr = sptr->next;

                if (fptr == sptr){
                    return true;
                }
            }      
        }
        
        return false;            
    }
};

File: LeetCode/Python/linkedlist-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: ListNode) -> bool:
        if head is not None and head.next is not None:
            fptr = head
            sptr = head
            
            while fptr and fptr.next:
                fptr = fptr.next.next
                sptr = sptr.next
                
                if fptr == sptr:
                    return True
                
            return False
